src/entry/process.ts

```diff
@@ -0,0 +1,110 @@
+import { stderr, stdout } from 'src/utils'
+import {
+  ReadyMessage,
+  RequestMessage,
+  ResponseMessage,
+  StartupMessage,
+  TinypoolWorkerMessage,
+} from '../common'
+import { getHandler, throwInNextTick } from './utils'
+
+type IncomingMessage =
+  | (StartupMessage & TinypoolWorkerMessage<'pool'>)
+  | (RequestMessage & TinypoolWorkerMessage<'port'>)
+
+type OutgoingMessage =
+  | (ReadyMessage & TinypoolWorkerMessage<'pool'>)
+  | (ResponseMessage & TinypoolWorkerMessage<'port'>)
+
+process.__tinypool_state__ = {
+  isChildProcess: true,
+  isTinypoolWorker: true,
+  workerData: null,
+  workerId: process.pid,
+}
+
+process.on('message', (message: IncomingMessage) => {
+  // Message was not for port or pool
+  // It's likely end-users own communication between main and worker
+  if (!message.__tinypool_worker_message__) return
```

Comment:
I think if we are concerned about this edge case, we should also check message type. I think you can send `null`/`undefined`?

Reply:
Now fixed. I was able to reproduce crash by sending `null` from worker and from main thread.